pylib/gyp/generator/compile_commands_json.py

```diff
@@ -108,10 +108,14 @@ def GenerateOutput(target_list, target_dicts, data, params):
         cwd = os.path.dirname(build_file)
         AddCommandsForTarget(cwd, target, params, per_config_commands)
 
+    output_dir = None
     try:
-        # generator_output can be `None` on Windows machines
-        output_dir = params["options"].generator_output or os.getcwd()
+        # generator_output can be `None` on Windows machines, or even not
+        # defined in other cases
+        output_dir = params["options"].generator_output
     except (AttributeError, KeyError):
```

Comment:
```suggestion
        output_dir = params.get("options").generator_output
    except AttributeError:
```

Reply:
We can also go with `getattr` if you really prefer the optionally get instead of handle exception approach.